src/software/thunderscope/gl/gl_hrvo_layer.py

```diff
@@ -0,0 +1,82 @@
+from pyqtgraph.Qt import QtCore, QtGui
+from pyqtgraph.opengl import *
+
+import numpy as np
+
+from proto.geometry_pb2 import Polygon
+from proto.visualization_pb2 import HRVOVisualization
+
+import software.thunderscope.constants as constants
+from software.py_constants import *
+from software.thunderscope.constants import Colors
+from software.thunderscope.thread_safe_buffer import ThreadSafeBuffer
+from software.thunderscope.gl.gl_layer import GLLayer
+from software.thunderscope.gl.graphics.gl_circle import GLCircle
+
+
+class GLHrvoLayer(GLLayer):
+    """GLHrvoLayer that visualizes the state of the HRVO Simulator"""
+
+    def __init__(self, name: str, robot_id: int, buffer_size: int = 5):
+        """Initialize the GLHrvoLayer
+
+        :param name: The displayed name of the layer
+        :param robot_id: The id of the robot which this layer will visualize
+        :param buffer_size: The buffer size, set higher for smoother plots.
+                            Set lower for more realtime plots. Default is arbitrary
+
+        """
+        GLLayer.__init__(self, name)
+
+        self.robot_id = robot_id
+        self.hrvo_buffer = ThreadSafeBuffer(buffer_size, HRVOVisualization)
+        self.prev_message = HRVOVisualization(robot_id=self.robot_id)
+
+        self.graphics_list.register_graphics_group(
+            "velocity_obstacles",
+            lambda: GLLinePlotItem(color=Colors.NAVIGATOR_OBSTACLE_COLOR),
+        )
+
+    def _update_graphics(self):
```

Comment:
I believe you're missing the drawings of the circles that represent the agents. Here's the missing part from the original hrvo layer:
```py
        for robot_circle in velocity_obstacle_msg.robots:
            painter.drawEllipse(
                self.createCircle(robot_circle.origin, robot_circle.radius)
            )
```

Reply:
fixed